tensorzero-core/src/db/clickhouse/query_builder/datapoint_queries.rs

```diff
@@ -0,0 +1,708 @@
+use serde::{Deserialize, Serialize};
+
+use crate::db::clickhouse::query_builder::parameters::add_parameter;
+use crate::db::clickhouse::query_builder::{ClickhouseType, QueryParameter, TagFilter, TimeFilter};
+
+/// Filter tree for querying datapoints.
+/// This is similar to `InferenceFilter` but without metric filters, as datapoints don't have associated metrics.
+#[cfg_attr(test, derive(ts_rs::TS))]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(test, ts(export))]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum DatapointFilter {
```

Comment:
can we open an issue to write a trait that unifies this logic with the Inference filtering logic?